### PR TITLE
mv: 'mv source hardlink' should fail

### PR DESCRIPTION
**Pull request by sylvestre**
fixes: tests/mv/force.sh

### CODE REVIEW

**Comment by sylvestre** [comment]
> Congrats! The gnu test tests/mv/force is no longer failing!
> Congrats! The gnu test tests/mv/i-4 is no longer failing!
> Congrats! The gnu test tests/mv/symlink-onto-hardlink is no longer failing!

Oh, it is unexpected :)

File: src/uucore/src/lib/features/fs.rs [hunk 1]
```diff
@@ -615,14 +615,47 @@ pub fn is_symlink_loop(path: &Path) -> bool {
...
+/// # Arguments
+///
+/// * `source` - A reference to a `Path` representing the source path.
+/// * `target` - A reference to a `Path` representing the target path.
+///
+/// # Returns
+///
+/// * `bool` - Returns `true` if the paths are hard links to the same file, and `false` otherwise.
+#[cfg(unix)]
+pub fn is_hardlink(source: &Path, target: &Path) -> bool {
```
Comment (cakebaker) [inline src/uucore/src/lib/features/fs.rs:635]: The function name doesn't really describe what the function does. Maybe something like `are_hardlinks_to_same_file` would be better?

**Review by cakebaker — COMMENTED** [review]
The functionality itself looks fine. However, there is a `unresolved import 'uucore::fs'` error when running `cargo test --features "mv" --no-default-features`. It can be fixed by adding `features=["fs"]` to mv's `Cargo.toml`.

**Comment by cakebaker** [comment]
I don't know if you have seen it, but the new tests in `test_mv.rs` fail on Android.

File: tests/by-util/test_mv.rs [hunk 2]
```diff
@@ -400,6 +400,40 @@ fn test_mv_same_file() {
...
+#[cfg(unix)]
+fn test_mv_same_hardlink() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    let file_a = "test_mv_same_file_a";
+    let file_b = "test_mv_same_file_b";
+    at.touch(file_a);
+
+    at.hard_link(file_a, file_b);
+
+    at.touch(file_a);
```
Comment (cakebaker) [inline tests/by-util/test_mv.rs:413]:
```suggestion
```
I don't know why this second "touch" is there. The test also passes without it.

File: tests/by-util/test_mv.rs [hunk 3]
```diff
@@ -400,6 +400,40 @@ fn test_mv_same_file() {
...
+#[cfg(unix)]
+fn test_mv_same_hardlink_backup_simple() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    let file_a = "test_mv_same_file_a";
+    let file_b = "test_mv_same_file_b";
+    at.touch(file_a);
+
+    at.hard_link(file_a, file_b);
+
+    at.touch(file_a);
```
Comment (cakebaker) [inline tests/by-util/test_mv.rs:430]:
```suggestion
```

**Pull request merged**